lib/propolis/src/vmm/hdl.rs

```diff
@@ -477,6 +629,24 @@ pub mod migrate {
         pub value: u64,
     }
 
+    #[derive(Copy, Clone, Debug, Default, Deserialize, Serialize)]
+    pub struct TimingInfoV1 {
+        /// guest TSC frequency (hz)
+        pub guest_freq: u64,
+
+        /// guest TSC
+        pub guest_tsc: u64,
+
+        /// monotonic host clock (ns)
+        pub hrtime: u64,
+
+        /// wall clock host clock
+        pub hrestime: Duration,
```

Comment:
I liked `Duration` because there are a bunch of existing functions for converting between it and nanoseconds, and for doing checked addition/subtraction to deal with overflow. Would you suggest something different?

Reply:
(If you think it would be better to use sec/ns as explicit fields in the serde type and convert internally to a Duration, that seems fine too)

Reply:
I would prefer explicit fields, since the internal fields of `Duration` are considered private.

As for internal calculations, using `Duration` for convenience and safety sounds ideal.

Reply:
done (`803081`)